pyat/at/tracking/track.py

```diff
@@ -1,73 +1,122 @@
+    # Split input in as many slices as processes
+    args = enumerate(numpy.array_split(r_in, pool_size, axis=1))
+    # Generate a new starting point for C RNGs
+    seed = random.common.integers(0, high=_imax, dtype=int)
+    global _globring
+    _globring = ring
+    if ctx.get_start_method() == 'fork':
+        passfunc = partial(_atpass_fork, seed, **kwargs)
+    else:
+        passfunc = partial(_atpass_spawn, ring, seed, **kwargs)
+    # Start the parallel jobs
+    with ctx.Pool(pool_size) as pool:
+        results = pool.starmap(passfunc, args)
+    _globring = None
+    # Gather the results
+    losses = kwargs.pop('losses', False)
+    return format_results(results, r_in, losses)
 
-        >>> @fortran_align
-        ... def element_pass(element: Element, r_in, **kwargs):
-        ... ...
 
-        Ensure than *r_in* is Fortran-aligned
-    """
-    @functools.wraps(func)
-    def wrapper(lattice, r_in, *args, **kwargs):
-        assert r_in.shape[0] == 6 and r_in.ndim in (1, 2), DIMENSION_ERROR
-        if r_in.flags.f_contiguous:
-            return func(lattice, r_in, *args, **kwargs)
-        else:
-            r_fin = numpy.asfortranarray(r_in)
-            r_out = func(lattice, r_fin, *args, **kwargs)
-            r_in[:] = r_fin[:]
-            return r_out
+@fortran_align
+def _element_pass(element: Element, r_in, **kwargs):
+    return _elempass(element, r_in, **kwargs)
+
 
-    return wrapper
+@fortran_align
+def _lattice_pass(lattice: list[Element], r_in, nturns: int = 1,
+                  refpts: Refpts = End, **kwargs):
+    refs = get_uint32_index(lattice, refpts)
+    kwargs['reuse'] = kwargs.pop('keep_lattice', False)
+    return _atpass(lattice, r_in, nturns, refpts=refs, **kwargs)
 
 
 @fortran_align
-def lattice_pass(lattice: Iterable[Element], r_in, nturns: int = 1,
-                 refpts: Refpts = End, **kwargs):
+def _plattice_pass(lattice: list[Element], r_in, nturns: int = 1,
+                   refpts: Refpts = End, pool_size: int = None,
+                   start_method: str = None, **kwargs):
+    refpts = get_uint32_index(lattice, refpts)
+    any_collective = has_collective(lattice)
+    kwargs['reuse'] = kwargs.pop('keep_lattice', False)
+    rshape = r_in.shape
+    if len(rshape) >= 2 and rshape[1] > 1 and not any_collective:
+        if pool_size is None:
+            pool_size = min(len(r_in[0]), multiprocessing.cpu_count(),
+                            DConstant.patpass_poolsize)
+        return _pass(lattice, r_in, pool_size, start_method, nturns=nturns,
+                     refpts=refpts, **kwargs)
+    else:
+        if any_collective:
+            warn(AtWarning('Collective PassMethod found: use single process'))
+        else:
+            warn(AtWarning('no parallel computation for a single particle'))
+        return _atpass(lattice, r_in, nturns=nturns, refpts=refpts, **kwargs)
+
+
+def lattice_track(lattice: Union[Element, Iterable[Element]], r_in,
```

Comment:
`lattice: Iterable[Element]` only, no `Union` any more. A single element is no more accepted